plugins/node/opentelemetry-instrumentation-restify/src/instrumentation.ts

```diff
@@ -25,25 +25,34 @@ import { VERSION } from './version';
 import * as constants from './constants';
 import {
   InstrumentationBase,
-  InstrumentationConfig,
   InstrumentationNodeModuleDefinition,
   InstrumentationNodeModuleFile,
   isWrapped,
+  safeExecuteInTheMiddle,
 } from '@opentelemetry/instrumentation';
 import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
 import { isPromise, isAsyncFunction } from './utils';
 import { getRPCMetadata, RPCType, setRPCMetadata } from '@opentelemetry/core';
+import type { RestifyInstrumentationConfig } from './types';
 
 const { diag } = api;
 
 export class RestifyInstrumentation extends InstrumentationBase<any> {
-  constructor(config: InstrumentationConfig = {}) {
+  constructor(config: RestifyInstrumentationConfig = {}) {
     super(`@opentelemetry/instrumentation-${constants.MODULE_NAME}`, VERSION);
```

Comment:
`config` need to be passed to `super` as well. Otherwise, it will not be available when set in constructor, only when set via `getConfig(config)` (as asserted in tests).

```js
super(`@opentelemetry/instrumentation-${constants.MODULE_NAME}`, VERSION, config);
```